adapters/handlers/rest/dynamic_user/handlers_dynamic_users.go

```diff
@@ -144,6 +152,10 @@ func (h *dynUserHandler) rotateKey(params users.RotateUserAPIKeyParams, principa
 		return users.NewRotateUserAPIKeyForbidden().WithPayload(cerrors.ErrPayloadFromSingleErr(err))
 	}
 
+	if h.staticUserExists(params.UserID) {
+		return users.NewRotateUserAPIKeyBadRequest().WithPayload(cerrors.ErrPayloadFromSingleErr(fmt.Errorf("user %v is static user", params.UserID)))
+	}
```

Comment:
Is this currently required because `user_id`s are equivalent to API keys? If that were to change, this requirement would be relaxed?

Reply:
> If that were to change, this requirement would be relaxed?

Then the question is how we would assign roles to static-api-key users.